Message/MessageFactory.php

```diff
@@ -88,6 +87,32 @@ protected function getDefaultOptions()
     }
 
     /**
+     * Find an email from database
+     *
+     * @param  string $reference
+     *
+     * @throws ReferenceNotFoundException
+     *
+     * @return  EmailInterface|null
+     */
+    public function getEmail($reference)
+    {
+        if (!isset($this->emails[$reference])) {
+            $this->emails[$reference] = $this->em->getRepository($this->options['email_class'])->findOneByReference($reference);
+        }
+
+        $email = $this->emails[$reference];
+
+        if (null === $email) {
+            throw new ReferenceNotFoundException($reference, sprintf('Reference "%s" does not exist for email.', $reference));
+        } elseif (!$email instanceof EmailInterface) {
```

Comment:
I wonder if this would ever get executed since it's coming from the repository.  Also, there's no message in the exception which would make it more difficult to debug if it ever were to get thrown. 

You could just  eliminate the first condition and just check instanceof because ```(null instanceof \Object)``` should always return false.

```php
if (!$email instanceof EmailInterface) {
    throw new ReferenceNotFoundException($reference, sprintf('Reference "%s" does not exist for email.', $reference));
}